### PR TITLE
capture_output

### PR DESCRIPTION
**Pull request by cccs-rs**
Modified code to use the capture_output flag in subprocess.run() to avoid duplicate logs written to stdout, stderr in the event of an exception

### CODE REVIEW

File: extract/extract.py [hunk 1]
```diff
@@ -668,10 +663,8 @@ def extract_zip(self, request: ServiceRequest, local: str, encoding: str):
                             try:
                                 shutil.rmtree(path, ignore_errors=True)
                                 os.mkdir(path)
-                                proc, _ = subprocess.Popen(
-                                    ['unrar', 'x', '-y', f'-p{password}', local, path],
-                                    env=env, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                                    stderr=subprocess.PIPE).communicate()
+                                proc = subprocess.run(['unrar', 'x', '-y', f'-p{password}', local, path],
```
Comment (cccs-kevin) [inline extract/extract.py:666]: nitpick: i think this should be renamed from proc -> stdout

File: extract/extract.py [hunk 2]
```diff
@@ -695,11 +688,8 @@ def extract_zip_7zip(self, request: ServiceRequest, local: str, encoding: str, p
         env['LANG'] = 'en_US.UTF-8'
 
         try:
-            stdoutput, stderr = subprocess.Popen(
-                ['7z', 'x', '-p', '-y', local, f'-o{path}'],
-                env=env, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE).communicate()
-            stdoutput += stderr
+            p = subprocess.run(['7z', 'x', '-p', '-y', local, f'-o{path}'], env=env, capture_output=True)
+            stdoutput = p.stdout + p.stderr
```
Comment (cccs-kevin) [inline extract/extract.py:692]: what are the odds that p.stderr will contain "Everything is OK"?

File: extract/extract.py [hunk 3]
```diff
@@ -709,11 +699,9 @@ def extract_zip_7zip(self, request: ServiceRequest, local: str, encoding: str, p
...
                         try:
                             shutil.rmtree(path, ignore_errors=True)
-                            stdoutput, stderr = subprocess.Popen(
-                                ['7za', 'x', f'-p{password}', f'-o{path}', local],
-                                env=env, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE).communicate()
-                            stdoutput += stderr
+                            p = subprocess.run(['7za', 'x', f'-p{password}', f'-o{path}', local],
+                                               env=env, capture_output=True)
+                            stdoutput = p.stdout + p.stderr
```
Comment (cccs-kevin) [inline extract/extract.py:704]: same question here

File: extract/extract.py (hunk 2 above)
Comment (cccs-rs) [reply extract/extract.py:692]: Likely none, but the messages like "Wrong Password" or "Can not open the file as archive" could appear in either.

Comment (cccs-kevin) [reply extract/extract.py:692]: ah, I didn't expand the code. Resolving!

**Review by cccs-kevin — APPROVED** [review]

**Pull request merged**